cmd/zpool/zpool_main.c

```diff
@@ -7140,6 +7146,9 @@ zpool_do_online(int argc, char **argv)
 				}
 			}
 		} else {
+			(void) fprintf(stderr, gettext("Failed to online "
+			    "\"%s\" in pool \"%s\": %d\n"),
+			    argv[i], poolname, rc);
```

Comment:
Isn't there a helper function that will parse the return value into a more helpful error string?

Reply:
Maybe, I'll look; this was originally written because someone was having an issue and the lack of any output at all was somewhat mystifying to both of us.